src/taipy/core/scenario/_scenario_model.py

```diff
@@ -30,37 +32,43 @@ class _ScenarioModel(_BaseModel):
         mapper_registry.metadata,
         Column("id", String, primary_key=True),
         Column("config_id", String),
-        Column("pipelines", JSON),
+        Column("tasks", JSON),
+        Column("additional_data_nodes", JSON),
         Column("properties", JSON),
         Column("creation_date", String),
         Column("primary_scenario", Boolean),
         Column("subscribers", JSON),
         Column("tags", JSON),
         Column("version", String),
+        Column("pipelines", JSON),
         Column("cycle", String),
     )
     id: ScenarioId
     config_id: str
-    pipelines: List[PipelineId]
+    tasks: List[TaskId]
+    additional_data_nodes: List[DataNodeId]
     properties: Dict[str, Any]
     creation_date: str
     primary_scenario: bool
     subscribers: List[Dict]
     tags: List[str]
     version: str
+    pipelines: Optional[List[PipelineId]] = None
     cycle: Optional[CycleId] = None
 
     @staticmethod
     def from_dict(data: Dict[str, Any]):
         return _ScenarioModel(
             id=data["id"],
             config_id=data["config_id"],
-            pipelines=data["pipelines"],
+            tasks=data.get("tasks", None),
+            additional_data_nodes=data.get("additional_data_nodes", None),
```

Comment:
Please make the migration at the lowest possible level, i.e., here.
You can have a look at the data node model module to have an example.